src/libmw/src/mmr/Segment.cpp

```diff
@@ -19,7 +19,7 @@ Segment SegmentFactory::Assemble(const IMMR& mmr, const ILeafSet& leafset, const
     while (segment.leaves.size() < num_leaves && leaf_idx < leafset.GetNextLeafIdx()) {
         if (leafset.Contains(leaf_idx)) {
             last_leaf_idx = leaf_idx;
-            segment.leaves.push_back(mmr.GetHash(leaf_idx.GetNodeIndex()));
```

Comment:
Isn't this supposed to store the leaf hashes? https://github.com/litecoin-project/litecoin/blob/970f4cf92c55a2a96f214b0f3a92529314e471ba/src/libmw/include/mw/mmr/Segment.h#L20-L21